### PR TITLE
Fix 32-bit overflow in Triton backward tensor offsets

### PR DESCRIPTION
**Pull request by huiyeruzhou**
## Summary

- promote flattened batch/head program ids to `tl.int64` in Triton backward kernels
- prevent base-pointer wraparound when a dense tensor exceeds `2^31` elements
- cover the main, split-launch, fused, and decode backward paths

## Minimal reproduction

Run on an A800 80GB with ffpa-attn 0.2.2, PyTorch 2.11.0, and Triton 3.6.0:

```python
import torch
from ffpa_attn import ffpa_attn_func

B, HQ, HKV = 24, 16, 2
NQ, NKV, D = 256, 12288, 512
q = torch.randn(B, HQ, NQ, D, device="cuda", dtype=torch.bfloat16, requires_grad=True)
k = torch.randn(B, HKV, NKV, D, device="cuda", dtype=torch.bfloat16, requires_grad=True)
v = torch.randn_like(k, requires_grad=True)
mask = torch.zeros(B, 1, NQ, NKV, device="cuda", dtype=torch.bfloat16)

out = ffpa_attn_func(
    q, k, v,
    attn_mask=mask,
    dropout_p=0.0,
    is_causal=False,
    enable_gqa=True,
    backend="triton",
)
out.mean().backward()
```

Forward succeeds, but backward raises `Triton Error [CUDA]: an illegal memory access was encountered`.

GQA backward expands K/V to the query-head layout. Here each expanded tensor contains `24 * 16 * 12288 * 512 = 2,415,919,104` elements, crossing `2^31`. The individual runtime strides still fit in i32, so multiplying the i32 program id by those strides wraps the base pointer.

## Verification

On the same A800:

- `B=16, NKV=12288`: still passes forward/backward
- `B=24, NKV=12288`: failed before, passes after the patch
- `B=24, NKV=24576`: failed before, passes after the patch; each expanded K/V tensor exceeds `2^32` elements

All patched runs produced finite outputs and Q/K/V gradients.

### CODE REVIEW

**Comment by huiyeruzhou** [comment]
CI note: the failing `build (3.12)` job stops in `actions/checkout@v4` before checking out or executing this PR. GitHub rejects the fork checkout because the workflow runs under `pull_request_target` with the base repository token (`Refusing to check out fork pull request code...`). This appears to be a base-workflow configuration issue rather than a code/test failure; a maintainer-triggered safe `pull_request` run or workflow adjustment is needed.

**Review by DefTruth — APPROVED** [review]

**Comment by DefTruth** [comment]
LGTM
